5.drug-dependency/scripts/4.drug_latent_dim_correlation.py

```diff
@@ -0,0 +1,263 @@
+                correlation_results.append(result_row)
+    
+    # Convert results into a dataframe
+    correlation_results_df = pd.DataFrame(correlation_results)
+    return correlation_results_df
+
+
+# In[3]:
+
+
+data_directory = pathlib.Path("../0.data-download/data").resolve()
+dependency_file = pathlib.Path(f"{data_directory}/CRISPRGeneEffect.parquet").resolve()
+gene_dict_file = pathlib.Path(f"{data_directory}/CRISPR_gene_dictionary.parquet").resolve()
+
+
+# In[4]:
+
+
+# Load PRISM data
+top_dir = "../5.drug-dependency"
+data_dir = "data"
+
+prism_df, prism_cell_df, prism_trt_df = load_utils.load_prism(
+    top_dir=top_dir,
+    data_dir=data_dir,
+    secondary_screen=False,
+    load_cell_info=True,
+    load_treatment_info=True,
+)
+
+# Reset the index and name it ModelID
+prism_df.reset_index(inplace=True)
+prism_df.rename(columns={'index': 'ModelID'}, inplace=True)
+prism_df.set_index('ModelID', inplace=True)
+prism_df.head()
+
+
+# In[5]:
+
+
+# Load metadata
+metadata_df_dir = pathlib.Path("../0.data-download/data/metadata_df.parquet")
+metadata = pd.read_parquet(metadata_df_dir)
+print(metadata.shape)
+
+#Load dependency data
+dependency_df, gene_dict_df = load_model_data(dependency_file, gene_dict_file)
+dependency_df.head()
+
+# Initialize the MinMaxScaler
+scaler = MinMaxScaler()
+
+# Apply the scaler to the numeric columns
+dependency_df[dependency_df.select_dtypes(include=['float64', 'int']).columns] = scaler.fit_transform(
+    dependency_df.select_dtypes(include=['float64', 'int'])
+)
+
+
+# In[6]:
+
+
+train_and_test_subbed_dir = pathlib.Path("../0.data-download/data/train_and_test_subbed.parquet")
+train_and_test_subbed = pd.read_parquet(train_and_test_subbed_dir)
+
+
+# Convert DataFrame to NumPy and then Tensor
+train_test_array = train_and_test_subbed.to_numpy()
+train_test_tensor = torch.tensor(train_test_array, dtype=torch.float32)
+
+#Create TensorDataset and DataLoader
+tensor_dataset = TensorDataset(train_test_tensor)
+train_and_test_subbed_loader = DataLoader(tensor_dataset, batch_size=32, shuffle=False)
+
+
+# In[7]:
+
+
+# Define the location of the saved models and output directory for correlation results
+model_save_dir = pathlib.Path("../4.gene_expression_signatures/saved_models")
+output_dir = pathlib.Path("results")
+output_dir.mkdir(parents=True, exist_ok=True)
+
+# Latent dimensions and model names to iterate over
+latent_dims = [2, 3, 4, 5, 6, 7, 8, 9, 10, 12, 14, 16, 18, 20, 25, 30, 35, 40, 45, 50, 60, 70, 80, 90, 100, 150, 200]
+model_names = ["pca", "ica", "nmf", "vanillavae", "betavae", "betatcvae"]
+
+# File to store the combined correlation results
+final_output_file = output_dir / "combined_latent_drug_correlations.parquet"
+try:
+    combined_results_df = pd.read_parquet(final_output_file)
+    print(f"Loaded existing results from {final_output_file}")
+except FileNotFoundError:
+    # If the file doesn't exist, initialize an empty DataFrame
+    combined_results_df = pd.DataFrame()
```

Comment:
Why do you want to load the existing file? Should these be overwritten? Yes?

Reply:
Also allows you to add latent dimensions or models without rerunning everything